src/main/io/serial.c

```diff
@@ -346,11 +346,6 @@ bool isSerialConfigValid(serialConfig_t *serialConfigToCheck)
             (portConfig->identifier == SERIAL_PORT_SOFTSERIAL2)) {
             // Ensure MSP or serial RX is not enabled on soft serial ports
             serialConfigToCheck->portConfigs[index].functionMask &= ~(FUNCTION_MSP | FUNCTION_RX_SERIAL);
-
-            // Ensure that the baud rate on soft serial ports is limited to 19200
-            serialConfigToCheck->portConfigs[index].gps_baudrateIndex = constrain(portConfig->gps_baudrateIndex, BAUD_AUTO, BAUD_19200);
```

Comment:
Don't delete this, just replace limit. User can see changed value (serial tab/cli) and can notice something is wrong. Without it, is will fail silently.

And for now only allow higher telemetry_baudrateIndex . 
Blackbox will be useless both at 19200 and 57600, but at 19200 there is good chance it transmis correct data.
GPS can be reconfigured to 19200, possibly starting on higher baudrate, so it may be allowed, with warning

Reply:
Had to remove the constrain code here for smartport to work. With in place it didn't work.